howso/ablation.amlg

```diff
@@ -109,6 +109,229 @@
+							; since a negative  % change needs to trigger the threshold.
+
+							;(current_index 1) is the top-level index of rel_threshold_map, which is
+							; comprised of the different measures (accuracy, precision, recall, rsmse... etc.)
+							; that have thresholds defined.
+							(if (contains_value (list "rmse" "r2") (current_index 1))
+								;Use (current_index 2) and (current_index 1) to access the values in the prediction stats
+								; maps, which have the same structure as the threshold map.
+								(/
+									(- 
+										(get new_prediction_stats_map (list (current_index 2) (current_index 1)))
+										(get prev_prediction_stats_map (list (current_index 2) (current_index 1)))
+									)
+									(get prev_prediction_stats_map (list (current_index 2) (current_index 1)))
+								)
+								(- (/
+									(- 
+										(get new_prediction_stats_map (list (current_index 2) (current_index 1)))
+										(get prev_prediction_stats_map (list (current_index 2) (current_index 1)))
+									)
+									(get prev_prediction_stats_map (list (current_index 2) (current_index 1)))
+								))
+							)
+							;(current_value) is the numerical value of the threshold. Exceeding this should satisfy the
+							; threshold and stop reduction.
+							(current_value)
+						)
+					)
+					;(current_value) contains the threshold map for a specific measure, e.g. for accuracy {target 0.05}
+					; would indicate a % change of greater than 5% would satisfy the threshold and stop reduction.
+					(current_value)
+				)
+			)
+			rel_threshold_map
+		)
+	)
+
+	;helper method which flattens the result of one of the !Check*Thresholds methods
+	; to determine whether at least one of the thresholds was violated (true) or not (false).
+	#!FlattenThresholdResultMap
+	(declare
+		(assoc
+			threshold_result_map (assoc)
+		)
+		(apply "or"
+			(values (map
+				(lambda
+					(apply "or" (values (current_value)))
+				)
+				threshold_result_map
+			))
+		)
+	)
+
+	;check all of the threshold maps against the current prediction stats map or the current
+	; and previous prediction stats maps. Returns the threshold map that failed.
+	;
+	;parameters:
+	; abs_threshold_map: a map of a measure to a map of feature name to absolute score threshold
+	; delta_threshold_map: a map of measure to a map of feature name to delta score threshold
+	; rel_threshold_map: a map of measure to a map of feature name to relative score threshold
+	; prev_prediction_stats_map: the previous prediction stats map.
+	; new_prediction_stats_map: the new prediction stats map.
+	#!CheckThresholds
+	(declare
+		(assoc
+			abs_threshold_map (assoc)
+			delta_threshold_map (assoc)
+			rel_threshold_map (assoc)
+			prev_prediction_stats_map (assoc)
+			new_prediction_stats_map (assoc)
+
+			;internal
+			abs_threshold_result_map (null)
+			delta_threshold_result_map (null)
+			output (assoc)
+		)
+
+		(assign (assoc
```

Comment:
I see 2 of these result_maps declared above, but not the third (the rel one). I think you could just not declare the two earlier and change this block to a (declare)